openpgp/packet/aead_encrypted.go

```diff
@@ -60,6 +60,9 @@ func (ae *AEADEncrypted) parse(buf io.Reader) error {
 	// Read initial nonce
 	mode := AEADMode(headerData[2])
 	nonceLen := mode.NonceLength()
+	if nonceLen < 12 {
+		return errors.AEADError("invalid nonce length")
+	}
```

Comment:
I would switch this around, because when this happens, it means that we don't recognize the AEAD mode, so as is this error would be misleading.
```suggestion
	if nonceLen == 0 {
		return errors.AEADError("unknown mode")
	}
```

Reply:
I thought about this, I nevertheless prefer using this same if to check for short nonces also. In the case `nonceLen == 0`, a quick debug will suffice to understand that the mode is unknown.

Reply:
But.. you're checking the return value of `mode.NonceLength()` here, not the length of the actual nonce in the packet. It is not possible for it to return a value less than 12 but greater than 0, and if it did, it would be because we added a mode that needs a shorter nonce, so we would change this check. So this error is always gonna mean the mode is unknown, and never that a mode has an invalid nonce length.

Reply:
Yes, I know it's not the length of the actual nonce in the packet. I was generally referring to using modified or new algorithms. The code does not look misleading to me (i.e., checking the nonce length and displaying "invalid nonce length" vs. checking the nonce length and displaying something related to the mode that depends on code inside the other function). However the behavior is the same, I'm committing your suggestion.

Reply:
Thanks :+1: Yeah, when looking at the new code in isolation it might look a bit strange indeed. It more accurately describes the cause of the error, though.